modules/guides/src/GuideStorage.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\guides;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Site\Settings;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Copyright (C) 2019 PRONOVIX GROUP BVBA.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+/**
+ * Provides GuideStorage to get the list of the guides.
+ */
+class GuideStorage extends ControllerBase {
+
+  /**
+   * The settings of the site.
+   *
+   * @var \Drupal\Core\Site\Settings
+   */
+  protected $settings;
+
+  /**
+   * GuideStorage constructor.
+   *
+   * @param \Drupal\Core\Site\Settings $settings
+   *   The settings of the site.
+   */
+  public function __construct(Settings $settings) {
+    $this->settings = $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('settings')
+    );
+  }
+
+  /**
+   * Get the directories from the guides directory.
+   *
+   * @return array
+   *   The array of the guide directories.
+   */
+  public function getDirs(): array {
+    $guides_dir = DRUPAL_ROOT . ($this->settings->get('guides_dir') ?? '/guides');
+    $dir = [];
+    foreach (array_diff(scandir($guides_dir), ['..', '.']) as $guide_dir) {
+      $path = $guides_dir . '/' . $guide_dir;
+      if (is_dir($path)) {
+        $dir[] = $guide_dir;
+      }
+    }
+
+    return $dir;
+  }
+
+  /**
+   * Get the path of the directories from the guides directory.
+   *
+   * @return array
+   *   The array of the path of the guide directories.
+   */
+  public function getDirPaths(): array {
```

Comment:
How this is different than the `getDirs()` or why it does not reuse what it returns?